core/kits/assets/js/globals/global-select-behavior.js

```diff
@@ -200,9 +205,25 @@ export default class GlobalControlSelect extends Marionette.Behavior {
 		if ( this.popover.isVisible() ) {
 			this.popover.hide();
 		} else {
-			this.popover.show();
+			if ( this.ui.$globalPreviewItemsContainer ) {
+				// This element is not defined when the controls popover is first loaded.
+				this.ui.$globalPreviewItemsContainer.remove();
+			}
 
-			this.setCurrentActivePreviewItem();
+			this.view.getGlobalsList()
+				.then(
+					( globalsList ) => {
+						// We just deleted the existing list of global preview items, so we need to rebuild it
+						// with the updated list of globals, register the elements and re-add the on click listeners.
+						this.addGlobalsListToPopover( globalsList );
```

Comment:
It is a different process between the first build and the re-build?
Can they use one function to do it?

Reply:
The first build only builds the header of the popover, without the globals list.
The globals list is built for the first time only on the first toggle click.